engine/access/rpc/backend/backend_network.go

```diff
@@ -109,6 +188,10 @@ func (b *backendNetwork) getValidSnapshot(snapshot protocol.Snapshot, blocksVisi
 	// Check if the counters and phase are different this indicates that the sealing segment
 	// of the snapshot requested spans either an epoch transition or phase transition.
 	if b.isEpochOrPhaseDifferent(counterAtHighest, counterAtLowest, phaseAtHighest, phaseAtLowest) {
+		if !findNextValidSnapshot {
+			return nil, SnapshotPhaseMismatchError
```

Comment:
```suggestion
			return nil, ErrSnapshotPhaseMismatch
```
nitpick: error naming convention (`ErrXYZ` for sentinel error values, `XYZError` for error types)